tools/python_integration_tests/classes/test.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2024 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import subprocess
+import time
+from collections import defaultdict
+import argparse
+from classes.ssh import SSHManager
+from classes.deployment import Deployment
+
+def run_command(cmd: str, err_msg: str = None) -> subprocess.CompletedProcess:
+        res = subprocess.run(cmd, shell=True, universal_newlines=True, check=True,
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if res.returncode != 0:
+            raise subprocess.SubprocessError(f"{err_msg}:\n{res.stderr}")
+        return res
+
+class Test:
+    # Base class for all tests.
+    def __init__(self, deployment):
+        self.deployment = deployment
+        self.sshmanager = None
+        self.ssh = None
+
+    def setup_deployment(self):
+        self.deployment.deploy()
+        
+        # Waiting to let the login node finish set up or ssh will fail.
+        time.sleep(60)
+
+class SlurmTest(Test):
+    # Base class for Slurm-specific tests.
+    def ssh_login(self):
+        self.sshmanager = SSHManager(self.deployment)
+        self.ssh = self.sshmanager.ssh_client
+        self.ssh.connect("localhost", port=10022, username=self.deployment.username, pkey=self.sshmanager.key)
+
+    def close_ssh(self):
+        self.sshmanager.close()
+
+    def assert_equal(self, value1, value2, message=None):
+        if value1 != value2:
+            if message is None:
+                message = f"Assertion failed: {value1} != {value2}"
+            raise AssertionError(message)
+
+    def get_nodes(self):
+        nodes = []
+        stdin, stdout, stderr = self.ssh.exec_command("scontrol show node| grep NodeName")
+        for line in stdout.read().decode().splitlines():
+            nodes.append(line.split()[0].split("=")[1])
+        return nodes
+
+class SlurmSimpleJobCompletionTest(SlurmTest):
```

Comment:
I believe this class declaration belongs `slurm_simple_job_completition.py`

Reply:
Done